### PR TITLE
Scheduler script

### PR DESCRIPTION
**Pull request by AbdoHamed-TheBadNomads**

### CODE REVIEW

File: Scheduler.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    AddSchedulerJobs(scheduler)
+    scheduler.start()
+
+def AddSchedulerJobs(scheduler):
+    scheduler.add_job(GetEndofMonthReport, "cron", day = int(os.getenv("End_of_Month_Report_Day")), hour = int(os.getenv("End_of_Month_Report_Hour")), minute = int(os.getenv("End_of_Month_Report_Minute")))
+    scheduler.add_job(GetEndofYearReport, "cron", month = int(os.getenv("End_of_Year_Report_Month")), day = int(os.getenv("End_of_Year_Report_Day")), hour = int(os.getenv("End_of_Year_Report_Hour")), minute = int(os.getenv("End_of_Year_Report_Minute")))
+    # add more jobs here
+
+async def GetEndofMonthReport():
+    finance_admin = await bot.fetch_user(int(os.getenv("Finance_Admin_id")))
```
Comment (AntWong-TheBadNomads) [inline Scheduler.py:22]: can we make this an array instead

File: Scheduler.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    global bot
+    bot = client
+    scheduler = AsyncIOScheduler(timezone = "Africa/Cairo")
+    AddSchedulerJobs(scheduler)
+    scheduler.start()
+
+def AddSchedulerJobs(scheduler):
+    scheduler.add_job(GetEndofMonthReport, "cron", day = int(os.getenv("End_of_Month_Report_Day")), hour = int(os.getenv("End_of_Month_Report_Hour")), minute = int(os.getenv("End_of_Month_Report_Minute")))
+    scheduler.add_job(GetEndofYearReport, "cron", month = int(os.getenv("End_of_Year_Report_Month")), day = int(os.getenv("End_of_Year_Report_Day")), hour = int(os.getenv("End_of_Year_Report_Hour")), minute = int(os.getenv("End_of_Year_Report_Minute")))
+    # add more jobs here
```
Comment (AntWong-TheBadNomads) [inline Scheduler.py:19]: remove comment

File: Scheduler.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import os
+import UI
+
+from datetime import datetime
+from apscheduler.schedulers.asyncio import AsyncIOScheduler
+from apscheduler.triggers.cron import CronTrigger
```
Comment (AntWong-TheBadNomads) [inline Scheduler.py:6]: where are we using this? is this import necessary?

File: Scheduler.py (hunk 1 above)
Comment (AbdoHamed-TheBadNomads) [reply Scheduler.py:22]: Done

File: Scheduler.py (hunk 2 above)
Comment (AbdoHamed-TheBadNomads) [reply Scheduler.py]: Done

File: Scheduler.py (hunk 3 above)
Comment (AbdoHamed-TheBadNomads) [reply Scheduler.py:6]: No, it was made for the old constructor but now we dont need it anymore

Comment (AbdoHamed-TheBadNomads) [reply Scheduler.py:6]: Removed

File: Scheduler.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    scheduler = AsyncIOScheduler(timezone = "Africa/Cairo")
+    AddSchedulerJobs(scheduler)
+    scheduler.start()
+
+def AddSchedulerJobs(scheduler):
+    scheduler.add_job(GetEndofMonthReport, "cron", day = int(os.getenv("End_of_Month_Report_Day")), hour = int(os.getenv("End_of_Month_Report_Hour")), minute = int(os.getenv("End_of_Month_Report_Minute")))
+    scheduler.add_job(GetEndofYearReport, "cron", month = int(os.getenv("End_of_Year_Report_Month")), day = int(os.getenv("End_of_Year_Report_Day")), hour = int(os.getenv("End_of_Year_Report_Hour")), minute = int(os.getenv("End_of_Year_Report_Minute")))
+
+async def GetEndofMonthReport():
+    finance_admins_ids = os.getenv("Finance_Admins_ids").split(",")
```
Comment (AntWong-TheBadNomads) [inline Scheduler.py:20]: trim spaces too please

File: Scheduler.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    for id in finance_admins_ids:
+        admin = await bot.fetch_user(int(id))
+        embed = UI.CreateGetEndOfMonthReportEmbed(member_db.GetMembers())
```
Comment (AntWong-TheBadNomads) [inline Scheduler.py:23]: make 1 embed for all admins, not 1 embed for each

File: Scheduler.py (hunk 4 above)
Comment (AbdoHamed-TheBadNomads) [reply Scheduler.py]: Done

File: Scheduler.py (hunk 5 above)
Comment (AbdoHamed-TheBadNomads) [reply Scheduler.py]: Done

File: Scheduler.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    bot = client
+    scheduler = AsyncIOScheduler(timezone = "Africa/Cairo")
+    AddSchedulerJobs(scheduler)
+    scheduler.start()
+
+def AddSchedulerJobs(scheduler):
+    scheduler.add_job(GetEndofMonthReport, "cron", day = int(os.getenv("End_of_Month_Report_Day")), hour = int(os.getenv("End_of_Month_Report_Hour")), minute = int(os.getenv("End_of_Month_Report_Minute")))
+    scheduler.add_job(GetEndofYearReport, "cron", month = int(os.getenv("End_of_Year_Report_Month")), day = int(os.getenv("End_of_Year_Report_Day")), hour = int(os.getenv("End_of_Year_Report_Hour")), minute = int(os.getenv("End_of_Year_Report_Minute")))
+
+async def GetEndofMonthReport():
```
Comment (AntWong-TheBadNomads) [inline Scheduler.py:19]:
```suggestion
async def SendEndofMonthReport():
```

File: Scheduler.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    finance_admins_ids = os.getenv("Finance_Admins_ids").split(", ")
+    embed = UI.CreateGetEndOfMonthReportEmbed(member_db.GetMembers())
+    for id in finance_admins_ids:
+        admin = await bot.fetch_user(int(id))
+        await admin.send(embed = embed)
+
+async def GetEndofYearReport():
```
Comment (AntWong-TheBadNomads) [inline Scheduler.py:26]:
```suggestion
async def SendEndofYearReport():
```

File: Scheduler.py (hunk 6 above)
Comment (AbdoHamed-TheBadNomads) [reply Scheduler.py]: Done

File: Scheduler.py (hunk 7 above)
Comment (AbdoHamed-TheBadNomads) [reply Scheduler.py]: Done

**Pull request merged**